soc/arm/arm/fvp_aemv8r_aarch32/arm_mpu_regions.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 IoT.bzh
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/slist.h>
+#include <linker/linker-defs.h>
+#include <arch/arm/aarch32/mpu/arm_mpu.h>
+
+#define DEVICE_REGION_START 0x80000000UL
+#define DEVICE_REGION_END   0xFFFFFFFFUL
+
+static const struct arm_mpu_region mpu_regions[] = {
+	/* Region 0 zephyr vector */
+	MPU_REGION_ENTRY("vector",
+			 (uintptr_t)_vector_start,
+			 REGION_RAM_TEXT_ATTR((uintptr_t)_vector_end)),
+	/* Region 1 zephyr text */
+	MPU_REGION_ENTRY("SRAM_0",
+			 (uintptr_t)__text_region_start,
+			 REGION_RAM_TEXT_ATTR((uintptr_t)__text_region_end)),
+
```

Comment:
Looks like there are sections between `__text_region_end` and `__rodata_region_start` when `CONFIG_CPLUSPLUS` enabled. https://github.com/zephyrproject-rtos/zephyr/blob/0521aa0fbce835022fbd65d0d438f4a20d5af893/include/arch/arm/aarch32/cortex_a_r/scripts/linker.ld#L158`

Reply:
does defining the end of the zephyr text region to (__rodata_region_start - 1), looks acceptable enough to not have to define all these sections indivdually ?

Reply:
yes, I am OK with it